reframe/frontend/statistics.py

```diff
@@ -86,6 +86,7 @@ def failure_report(self):
 
             report.append(line_width * '-')
             report.append('FAILURE INFO for %s %s' % (check.name, retry_info))
+            if check.descr: report.append('  * Test Description: %s' % check.descr)
```

Comment:
Just print the description unconditionally. `check.descr` will always be non-empty currently, since by default it's the test's name.